hydromt_wflow/workflows/rootzoneclim.py

```diff
@@ -0,0 +1,857 @@
+        Xarray datarray containing the future aridity index values per sub 
+        catchment.
+
+    Returns
+    -------
+    xarray datarray containing the future discharge coefficient.
+    """  
+    return - (aridity_index - (1 + aridity_index**omega)**(1/omega))
+
+
+def check_inputs(start_hydro_year,
+                 start_field_capacity,
+                 dsrun,
+                 ds_obs,
+                 ds_cc_hist,
+                 ds_cc_fut):
+    
+    # Start with some initial checks
+    list_of_months = ["Jan", 
+                      "Feb", 
+                      "Mar", 
+                      "Apr", 
+                      "May", 
+                      "Jun", 
+                      "Jul", 
+                      "Aug", 
+                      "Sep", 
+                      "Oct", 
+                      "Nov", 
+                      "Dec"]
+    if start_hydro_year not in list_of_months:
+        raise ValueError(
+            f"start_hydro_year not in {list_of_months}: provide a valid month"
+            )
+    if start_field_capacity not in list_of_months:
+        raise ValueError(
+            f"start_field_capacity not in {list_of_months}: provide a valid month"
+            )
+    
+    if "run" not in list(dsrun.keys()):
+        raise ValueError(
+            "Variable run not in run_fn"
+            )
+    
+    if "precip" not in list(ds_obs.keys()):
+        raise ValueError(
+            "Variable precip not in forcing_obs_fn"
+            )
+
+    if "pet" not in list(ds_obs.keys()):
+        raise ValueError(
+            "Variable pet not in forcing_obs_fn"
+            )
+    if ds_cc_hist != None:
+        if "precip" not in list(ds_cc_hist.keys()):
+            raise ValueError(
+                "Variable precip not in forcing_cc_hist_fn"
+                )
+    
+        if "pet" not in list(ds_cc_hist.keys()):
+            raise ValueError(
+                "Variable pet not in forcing_cc_hist_fn"
+                )
+    if ds_cc_fut != None:
+        if "precip" not in list(ds_cc_fut.keys()):
+            raise ValueError(
+                "Variable precip not in forcing_cc_fut_fn"
+                )
+    
+        if "pet" not in list(ds_cc_fut.keys()):
+            raise ValueError(
+                "Variable pet not in forcing_cc_fut_fn"
+                )
+            
+    return None
+
+
+def rootzoneclim(ds_obs,
+                 ds_cc_hist,
+                 ds_cc_fut,
+                 dsrun, 
+                 ds_like, 
+                 flwdir, 
+                 return_period,
+                 Imax, 
+                 start_hydro_year,
+                 start_field_capacity,
+                 LAI,
+                 rooting_depth,
+                 correct_cc_deficit,
+                 chunksize,
+                 missing_days_threshold,
+                 logger=logger):
+    """
+    Returns root zone storage parameter for current observed and (optionally 
```

Comment:
Could you maybe then base this on the current wflow forcing file ? Or external data? Does anything happen if you have the three forcing available or do you just add a layer per forcing? If the later the user could also just call the function three times. This would reduce complexity

Reply:
not sure I understand what you mean here. If you want to calculate the rootzone storage capacity for the future, you need to use results of ds_obs and potentially ds_cc_ref, so it wouldn't be possible to call the function 3 times with a different ds_obs. otherwise, it would indeed decrease complexity...